packages/flutter/lib/src/widgets/form.dart

```diff
@@ -293,19 +293,45 @@ class FormState extends State<Form> {
   /// returns true if there are no errors.
   ///
   /// The form will rebuild to report the results.
+  /// See also: [validateGranually]
   bool validate() {
     _hasInteractedByUser = true;
     _forceRebuild();
-    return _validate();
+    return _validate().isValid;
   }
 
-  bool _validate() {
+
+  /// Validates every [FormField]s that is a descendant of this [Form], and
+  /// returns a [Map] which keys' are [Key]s of the descendant [FormField]s
+  /// and values' are their corresponding validation results.
+  ///
+  /// Any [FormField] that has not been asigned a key will be ignored in the result.
+  /// To get the absolute validation value of the Form, Consider using [validate]
+  ///
+  /// Common usage of this method is when you need draw the user's attention
+  /// to the invalid field(s) using their widget key(s).
+  ///
+  /// The form will rebuild to report the results.
+  Map<Key, bool> validateGranually() {
+    _hasInteractedByUser = true;
+    _forceRebuild();
+    return _validate().fieldsValidationStatus;
+  }
+
+  _FormValidationStatus _validate() {
+    final Map<Key, bool> fieldsValidationStatus = <Key, bool>{};
     bool hasError = false;
     String errorMessage = '';
     for (final FormFieldState<dynamic> field in _fields) {
       hasError = !field.validate() || hasError;
       errorMessage += field.errorText ?? '';
+      final Key? key = field.widget.key;
```

Comment:
what's the reason you want to return a `Key` instead of the field itself, and only include keyed fields in the returned `Map`?